django/contrib/postgres/fields/array.py

```diff
@@ -29,9 +29,17 @@ def __init__(self, base_field, size=None, **kwargs):
             self.default_validators.append(ArrayMaxLengthValidator(self.size))
         super(ArrayField, self).__init__(**kwargs)
 
-    def contribute_to_class(self, cls, name, **kwargs):
-        super(ArrayField, self).contribute_to_class(cls, name, **kwargs)
-        self.base_field.model = cls
+    def _get_model(self):
+        try:
+            return self.__dict__['model']
+        except KeyError:
+            raise AttributeError
+
+    def _set_model(self, model):
+        self.__dict__['model'] = model
+        self.base_field.model = model
+
+    model = property(_get_model, _set_model)
```

Comment:
Could do. I usually avoid it in my own code because while the `super()` getter is easily accessible with `super().attribute` you to have deal with the `property` internals (`fset`) to access the setter.